lib/src/utils.dart

```diff
@@ -37,19 +38,46 @@ final String _sdkDir = (() {
 })();
 
 final bool isFlutterSdk = (() {
-  const depth = 7;
   final components = p.split(Platform.resolvedExecutable);
-  if (components.length < depth) {
+  return isFlutterSdkHeuristic(components);
+})();
+
+@visibleForTesting
```

Comment:
This isn't exported anywhere, right? I don't think this is needed.

Reply:
Not exported. I'm just used to do this in order to mark a member, even internally inside a package, as "private". (I.e. no other files in the package should access this.) `visibleForTesting` does this: it'll show a lint warning when you try to use the member anywhere except this file itself or a file in `test/`.

If you want, I can remove it, though. Matter of taste.